allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -493,6 +516,18 @@ def _download_file(self, file_attributes: CacheFileAttributes) -> bool:
 
         version_id = file_attributes.version_id
 
+        pbar = None
+        if not self._file_exists(file_attributes):
+            response = self.s3_client.list_object_versions(Bucket=bucket_name,
+                                                           Prefix=str(obj_key))
+            object_info = [i for i in response["Versions"]
+                           if i["VersionId"] == version_id][0]
+            pbar = tqdm.tqdm(desc=object_info["Key"].split("/")[-1],
+                             total=object_info["Size"],
+                             unit_scale=True,
+                             unit_divisor=1000.,
+                             unit="MB")
+
         while not self._file_exists(file_attributes):
             response = self.s3_client.get_object(Bucket=bucket_name,
```

Comment:
Is `self.s3_client.get_object()` blocking?

Reply:
I think you get a response with `Body` key and that is a stream that can be read.
@danielsf might have more insight, as this was his implementation originally.